packages/stores/src/storeProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { PropsWithChildren, useEffect, useMemo } from 'react';
+import StoreContext from './storeContext';
+import { ExchangeRatesStore } from './stores';
+import { ExchangeRatesProvider } from './providers';
+import type { TRootStore, TStores } from '../types';
+
+const StoreProvider = ({ children, store }: PropsWithChildren<{ store: TRootStore }>) => {
+    const memoizedValue: TStores = useMemo(
+        () => ({
+            ...store,
+            exchange_rates: new ExchangeRatesStore(),
+        }),
+        [store]
+    );
+
+    useEffect(() => {
+        return () => {
+            memoizedValue.exchange_rates.unmount();
+        };
+    }, [memoizedValue]);
+
+    return (
+        <StoreContext.Provider value={memoizedValue}>
+            <ExchangeRatesProvider>{children}</ExchangeRatesProvider>
```

Comment:
In the future, if we add more subscription providers, we will be led to react context hell.

Can we consider this? - https://dev.to/hyposlasher/no-more-react-context-hell-4pm3

Reply:
@balakrishna-deriv as I spoke with @farzin-deriv , he told me that we'll find a solution for this later.